kubemarine/procedures/upgrade.py

```diff
@@ -137,6 +138,22 @@ def upgrade_containerd(cluster: KubernetesCluster):
                 if isinstance(value, dict):
                     config_string += f"\n[{key}]\n{toml.dumps(value)}"
             utils.dump_file(cluster, config_string, 'containerd-config.toml')
+            
+            node_group=cluster.nodes['control-plane'].include_group(
+                        cluster.nodes.get('worker')).get_ordered_members_list(
+                        provide_node_configs=True)
+            for node in node_group:
+                if "control-plane" in node["roles"]:
+                    control_plane = node
+                else:
+                    control_plane = cluster.nodes["control-plane"].get_first_member(provide_node_configs=True)
+                kubeadm_flags_file = "/var/lib/kubelet/kubeadm-flags.env"
+                pause_version = cluster.globals['compatibility_map']['software']['pause'][target_kubernetes_version]['version']
+                kubeadm_flags = node['connection'].sudo(f"cat {kubeadm_flags_file}",
+                                        is_async=False).get_simple_out()
+                updated_kubeadm_flags = re.sub(r'pause:.*\..*', f"pause:{pause_version}", kubeadm_flags)
+                node['connection'].put(StringIO(updated_kubeadm_flags), kubeadm_flags_file, backup=True, sudo=True)
```

Comment:
After the task is finished, /usr/bin/kubelet is still running with old --pod-infra-container-image. Also I am concerned about that already existing containers are still created using the old image, but looks like this is an overall gap of the task.

Reply:
command to restart kubelet service has been added.